### ISSUE TITLE
Update asset is prone to race condition

### ISSUE THREAD

**Issue by Aschen** [kuzzleio/kuzzle-device-manager#216]
# Description

We need to update the whole metadata array when updating assets document so when metadata are partially updated, we need to keep the previous metadata and merge with the new one.

2 calls to `update` will be vulnerable to race condition since we are getting the asset, merging metadata, emitting event and finally updating.

**Comment by ChillPC** [comment]
Isn't the whole manipulation of devices and assets vulnerable to race condition since we do not create mutex on top of ES, would it be linking, adding measures...

**Comment by TisserantG** [comment]
No, it's because two element try to alter content of the same field. (not erased it, but alter it)
In other update, that does nos cause race condition because the field is erased and we do not use the previous value to create the new. 
We can probably using https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#upserts to resolve the problem.
